src/engine/SpatialJoin.cpp

```diff
@@ -312,23 +315,30 @@ VariableToColumnMap SpatialJoin::getVarColMapPayloadVars() const {
                     "column map can be computed.");
 
   auto payloadVariablesVisitor = [this]<typename T>(const T& value) {
-    auto varColMap =
-        childRight_->getRootOperation()->getExternallyVisibleVariableColumns();
+    auto varColMap = childRight_->getVariableColumns();
     VariableToColumnMap newVarColMap;
 
     if constexpr (std::is_same_v<T, PayloadAllVariables>) {
       newVarColMap = VariableToColumnMap{varColMap};
     } else {
       static_assert(std::is_same_v<T, std::vector<Variable>>);
-      for (auto var : value) {
+      for (const auto& var : value) {
+        AD_CONTRACT_CHECK(varColMap.contains(var), [&]() {
+          return absl::StrCat("Variable '", var.name(),
+                              "' selected as payload to spatial join but not "
+                              "present in right child.");
+        });
```

Comment:
The problem with this solution is that `addWarningOrThrow` may not be const because it must be able to modify the `warnings_` attribute. However `computeVariableToColumnMap` which needs to use this function is const and therefore cannot be overridden as non-const.